model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro.py

```diff
@@ -895,12 +896,12 @@ def time_step(
 
         self._update_theta_v(
             mask_prog_halo_c=self._metric_state_nonhydro.mask_prog_halo_c,
-            rho_now=prognostic_state_ls[nnow].rho,
-            theta_v_now=prognostic_state_ls[nnow].theta_v,
-            exner_new=prognostic_state_ls[nnew].exner,
-            exner_now=prognostic_state_ls[nnow].exner,
-            rho_new=prognostic_state_ls[nnew].rho,
-            theta_v_new=prognostic_state_ls[nnew].theta_v,
+            rho_now=prognostic_state_swp.current.rho,
+            theta_v_now=prognostic_state_swp.current.theta_v,
+            exner_new=prognostic_state_swp.other.exner,
```

Comment:
Since `Swapping` is part of the application code, we could use `current` and `next`  or even `now` and `next` instead of `current` and `other`. 
`other` is perfect for a generic `Swapping` but less readable in the application code where this is used for timestepping.

Reply:
Refactored `Swapping` to generic `Pair` which can be easily subclassed to set accessors with custom names.